app/views/layouts/_asset_reconciliation.html.erb

```diff
@@ -0,0 +1,60 @@
+<meta name="page-cached-at" content="<%= Time.current.to_i %>">
+
+<meta name="proper-stylesheet-crayons" content="<%= stylesheet_url "crayons" %>">
+<meta name="proper-stylesheet-views" content="<%= stylesheet_url "views" %>">
+<meta name="proper-stylesheet-minimal" content="<%= stylesheet_url "minimal" %>">
+
+<script async>
+
+  // This area is designed to reconcile issues of mismatched caches stylesheets.
+  // Currently it handles a scenario where the head assets are out of date and need to be updated.
+  // There is a legitimate reverse scenario we do not yet handle where we *should* use the "out-of-date"
+  // content on the page, but since we may have issues persisting old stylesheets etc.
+  // this only handles the most troubling issues of pages that load with no styles.
+
+  var headCacheCheck = document.querySelector("meta[name='head-cached-at']");
+  var pageCacheCheck = document.querySelector("meta[name='page-cached-at']");
+  var headCrayons = document.getElementById("main-crayons-stylesheet")
+  var headViews = document.getElementById("main-views-stylesheet")
+  var headMinimal = document.getElementById("main-minimal-stylesheet")
+  var pageCrayons = document.querySelector("meta[name='proper-stylesheet-crayons']");
+  var pageViews = document.querySelector("meta[name='proper-stylesheet-views']");
+  var pageMinimal = document.querySelector("meta[name='proper-stylesheet-minimal']");
+
+  if (headCacheCheck && headCrayons && // If these elements exist, we can proceed.
```

Comment:
If the page is more current than the head _and_ we have any assets which don't match up with the ones in the head, we add new stylesheets and get rid of the old ones.